src/codeu/chat/server/Server.java

```diff
@@ -206,7 +246,125 @@ public void run() {
         timeline.scheduleIn(RELAY_REFRESH_MS, this);
       }
     });
-    
+  }
+
+  // This method will write the contents of dataList to a txt file
+  public void saveServer() {
+    timeline.scheduleIn(SAVE_SERVER_MS, new Runnable() {
+      @Override
+      public void run() {
+        try {
+          // Check if the data storage file exists, and if not, create it
+          if (!dataStorage.exists()) {
+            dataStorage.createNewFile();
+            LOG.info("Created data storage file.");
+          }
+
+          fileWriter = new FileWriter(dataStorage, true);
+          bufferedWriter = new BufferedWriter(fileWriter);
+
+          LOG.info("Writing sever content.");
+
+          for (int i = 0; i < dataList.size(); i++) {
+            // Write JSON object with appended identifer to file
+            // Ends up looking like: <Identifier>;<JSON Object> on every line
+            bufferedWriter.write(dataList.get(i));
+            bufferedWriter.newLine();
+          }
+        } catch (IOException ex) {
+          LOG.error(ex, "There was an exception while writing server content.");
+        } finally {
+          try {
```

Comment:
The cleanup of bufferedWriter, fileWriter is repeated in 2 places. Consider refactoring that into a cleanup method that you can invoke from both the places.